catroid/src/main/java/org/catrobat/catroid/camera/CameraManager.java

```diff
@@ -96,8 +96,14 @@ public static CameraManager getInstance() {
 		return instance;
 	}
 
+	public static void setInstance(CameraManager cameraManager) {
+		instance = cameraManager;
+	}
+
 	public static void makeInstance() {
-		instance = new CameraManager();
+		if (instance == null) {
```

Comment:
I agree with this, because the constructor does some work & it's unnecessary to do it for each camera permission.
There are tests that possibly rely on makeInstance() resetting the CameraManager.
I have not had time to look into it further, but it's possible that the Stage too relies on CameraManager having a clean state. Check this out and see if it's worth it!